pkg/backend/filestate/state_test.go

```diff
@@ -0,0 +1,61 @@
+package filestate
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gocloud.dev/blob/memblob"
+)
+
+func TestIsStateEmpty(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		desc string
+		// List of files that exist in the bucket.
+		files []string
+
+		// Whether the state is considered empty.
+		empty bool
+	}{
+		{
+			desc:  "empty",
+			empty: true,
+		},
+		{
+			desc: "non-state files",
+			files: []string{
+				"foo",
```

Comment:
Nit: I think it could be a neat example to include the pulumi meta file in an this example.